File: addrev.py
import math

n = int(input()) 
resp = []

def addrev(t:int)->int: 
	k = math.ceil( math.log10(t) ) 
	rv = 0
	u = t

	if t != 10**k : 
		k = k -1 
	else: 
		return t//10**k

	if k == 0: 
		return t 
	else: 
		while k != 0: 
			rv = rv + (u%10)*10**k 
			u = u//10
			k = k - 1
			
		return rv + u
			
for i in range(n):  
	arr = input().split()
	t1 = addrev( int(arr[0]) ) 
	t2 = addrev( int(arr[1]) ) 
	resp.append( addrev( t1 + t2 ) )
	
	
for i in resp: print(i) 
	

File: basic/addrev.cpp
#include<iostream> 
#include<stdio.h> 
#include<math.h>
#include<string> 
	
using namespace std; 
 
int reverse(int number);
 
int main() 
{
	int resp[10000]; 
	int n, a, b, temp; 
	cin>>n; 
	
	for (int i=0; i<n; i++) {
		cin>>a>>b; 
		temp = reverse(a) + reverse(b); 
		resp[i] = reverse(temp);
	}
			
	for(int j=0; j<n; j++) { 
		cout<<resp[j]<<endl; 
	}
	
	return 0; 
}
 
int reverse(int number){
 
	if (number<10)
	{
		return number; 
	}
 
	int aux, c, digit;
	aux = c = 0;
 
	int stop = ceil( log10(number) ); 
	stop += (number/pow(10,stop)==1)? 1: 0; 
		
	stop--;
 
	while (stop!=-1)
	{
		digit = floor(number/pow(10,stop));
		aux += digit*pow(10,c);
		number -= digit*pow(10,stop); 	
		stop--;
		c++;
	}
	
	return aux; 
}
